dagshub/data_engine/model/datasource.py

```diff
@@ -1200,6 +1211,73 @@ def contains(self, item: str):
         self._test_not_comparing_other_ds(item)
         return self.add_query_op("contains", item)
 
+    def _periodic_filter(self, item: Union[str, List[str]], periodtype):
+        self._test_not_comparing_other_ds(item)
+        self._query.timezone = _get_local_timezone()
+        return self.add_query_op(periodtype, item)
```

Comment:
1. yes, by default we use local tz. unless user explicitly uses time_zone()
2. comparing what? you mean verifying that year/month/day filters are only used on time fields?

Reply:
1. What I mean is that if I run `ds.time_zone()` and then run any of the filters, it looks like it will actually reset to the local.
2. I meant making sure that the filters throw errors when used on a non-datetime, yes. Honestly I'm w/e on it, you can skip it

Another thing for naming: `time_zone()` -> `with_time_zone()`